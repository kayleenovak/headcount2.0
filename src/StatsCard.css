.stat-card {
  border-radius: 10px;
  box-shadow: 2px 2px 5px black;
  padding: 10px;
  background: white;
  height: 300px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 250px;
  margin: 0 20px 0 20px;
}