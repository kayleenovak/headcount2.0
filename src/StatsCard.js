import React from 'react';
import './StatsCard.css'

const StatsCard = (props) => {
  return (
    <div>
    {
      props.schoolsForComparison.length > 1 ? 
      <div className='stat-card'>
        <h3>{ props.schoolsForComparison[0] } average: { props.comparison[props.schoolsForComparison[0]] }</h3>
        <h2>Both schools compared: { props.comparison.compared }</h2>
        <h3>{ props.schoolsForComparison[1] } average: { props.comparison[props.schoolsForComparison[1]]}</h3>
      </div> : 
      null
    }
    </div>
  )
}

export default StatsCard