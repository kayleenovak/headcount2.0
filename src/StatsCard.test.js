import React from 'react';
import { shallow, mount } from 'enzyme'
import StatsCard from './StatsCard.js'

describe('StatsCard', () => {
  it('should match the snapshot', () => {
    const mockSchools = ['COLORADO', 'COLORADO SPRINGS 11']
    const mockComparison = {
      'COLORADO': 1,
      'COLORADO SPRINGS 11': 1,
      'compared': 1
    }
    const wrapper = shallow(<StatsCard schoolsForComparison={ mockSchools } comparison={ mockComparison }/>)

    expect(wrapper).toMatchSnapshot()
  })
})