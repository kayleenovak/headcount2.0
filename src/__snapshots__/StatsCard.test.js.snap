// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsCard should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatsCard
    comparison={
        Object {
            "COLORADO": 1,
            "COLORADO SPRINGS 11": 1,
            "compared": 1,
          }
    }
    schoolsForComparison={
        Array [
            "COLORADO",
            "COLORADO SPRINGS 11",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <h3>
                COLORADO
                 average: 
                1
        </h3>
        <h2>
                Both schools compared: 
                1
        </h2>
        <h3>
                COLORADO SPRINGS 11
                 average: 
                1
        </h3>
</div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            COLORADO
             average: 
            1
</h3>,
          <h2>
            Both schools compared: 
            1
</h2>,
          <h3>
            COLORADO SPRINGS 11
             average: 
            1
</h3>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO",
              " average: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO",
            " average: ",
            1,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Both schools compared: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Both schools compared: ",
            1,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "COLORADO SPRINGS 11",
              " average: ",
              1,
            ],
          },
          "ref": null,
          "rendered": Array [
            "COLORADO SPRINGS 11",
            " average: ",
            1,
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <h3>
                    COLORADO
                     average: 
                    1
          </h3>
          <h2>
                    Both schools compared: 
                    1
          </h2>
          <h3>
                    COLORADO SPRINGS 11
                     average: 
                    1
          </h3>
</div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              COLORADO
               average: 
              1
</h3>,
            <h2>
              Both schools compared: 
              1
</h2>,
            <h3>
              COLORADO SPRINGS 11
               average: 
              1
</h3>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COLORADO",
                " average: ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              "COLORADO",
              " average: ",
              1,
            ],
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Both schools compared: ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Both schools compared: ",
              1,
            ],
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "COLORADO SPRINGS 11",
                " average: ",
                1,
              ],
            },
            "ref": null,
            "rendered": Array [
              "COLORADO SPRINGS 11",
              " average: ",
              1,
            ],
            "type": "h3",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
